### PR TITLE
roe-2599 disable validation filters in validateFull

### PR DESCRIPTION
**Pull request by lduranteau**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2599

### Change description
disable validation filters in validateFull

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:927]: is this test already performed by the one above (@ValueSource(booleans = { false, true }))

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:917]: i think the is_secure_register should return error if it is 'true'

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:917]: ah, feature flag false so should be no errors

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:947]: the test name indicates that an error is reported, but the test checks for no errors

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:1028]: test name references FullValidation but I think it should be PartialValidation (test uses validatePartial)

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:927]: you are right, forgot to remove it

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:947]: I will update the name

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:1028]: I will update the name

**Review by dlloyd1-ch — APPROVED** [review]
LGTM. Worth getting another approval off Mark or Martin before merging.

**Comment by dlloyd1-ch** [comment]
Also, we can't merge this until the api code freeze is lifted.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/OverseasEntitySubmissionDto.java
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/OverseasEntitySubmissionDto.java:263]: Looks like code used to convert the supplied value to uppercase but no longer. Wondering why this was done and why it's being removed - there's no mention of this in the JIRA ticket.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java
Comment (mwestacott) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:129]: I feel we should have a comment somewhere (perhaps around line 74 in the `validateFull()` method of this class?) just stating that fields used **only** for web navigation purposes are deliberately not validated by the API.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/OverseasEntitySubmissionDto.java
Comment (lduranteau) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/dto/OverseasEntitySubmissionDto.java:263]: The code that was here has been moved into the validator.

File: src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java
Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:930]: The full validation never calls these methods so there is no point in adding tests to check that the methods that are not called don't raise an error.

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:930]: I agree with Martin, I think it would be good to have that test as we are testing that the data in that state should pass validation, and if someone would change the behaviour of validation to not allow nulls for those fields then the test would catch it before it goes into production.

Comment (lduranteau) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:930]: validateFull never uses the validateHasSoldLand, validateIsSecureRegister and validateWhoIsRegistering methods.
So if the behaviour changes in these methods, it will have no impact, as they are not called from validateFull.

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:930]: The new test would be for the requirement/use case that the full validation should allow nulls for those 3 fields. If someone were to add the null validation for those 3 fields into validateFull (as they are unaware of our discussions) then I don't think a test would fail and it would appear that everything is working as expected.

Comment (mwestacott) [reply src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidatorTest.java:930]: Ok, discussed this. Agreed that the comment in the application code will suffice here.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java
Comment (mwestacott) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntitySubmissionDtoValidator.java:129]: Discussed this. Reason is to document the difference in validation behaviour for 'future generations' 😄.

Suggest we add this as the first line in the `validateFullRegistrationDetails()` method:

// Note that model fields used purely to support web journey navigation are not submitted with the filing and therefore aren't checked (e.g. isSecureRegister)

@markpit @lduranteau sound ok?

**Review by markpit — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**
